azavea/raster-foundry
# Add import functionality for S3-hosted GeoTiffs

## Overview

This adds import functionality to the `rf` module for creating Scenes from GeoTIFFs hosted in an S3 folder. It specifically targets 4-band imagery taken with the [UltraCam Falcon](http://www.vexcel-imaging.com/products/ultracam-falcon/) since this is a camera commonly used by aerial imagery providers. This will eventually need to be generalized to support the UltraCam Falcon's panchromatic imagery so that we can support pan-sharpening.

### Checklist


### Notes

- GeoTIFFs can support resolution in both the X and Y directions; in other words, they support pixels that are rectangular with respect to the ground. Our API doesn't support this yet, and in this case the X and Y resolutions are the same, so I've just used the X resolution here, but we may want to consider supporting this in the future if we encounter GeoTIFFs where the X and Y resolutions are not the same.
- I've made guesses about the widths of each frequency band based on an assumption that UltraCams use color filters with similar characteristics to those used in the Bayer filters in standard digital cameras. However, this may not be accurate; it would be good to get real numbers (e.g. an absorption graph) for the R, G, B, NIR, and panchromatic bands.

## Testing Instructions

 * `./scripts/console airflow-worker bash`
 * `ipython`
 * 
```
from rf.uploads.geotiff import GeoTiffS3SceneFactory
import datetime
from rf.utils.io import Visibility

captureDate = datetime.date(2016, 8,1)
scene = None
factory = GeoTiffS3SceneFactory('dfac6307-b5ef-43f7-beda-b9f208bb7726', Visibility.PRIVATE, 'test-data-source', captureDate, 'rasterfoundry-staging-data-us-east-', '<kayak imagery folder prefix here>')
for s in factory.generate_scenes():
    scene = s
    break
```
* `scene.to_dict()` and verify that everything looks as expected.

Closes #942
